functions/go/project-service-list/README.md

```diff
@@ -0,0 +1,117 @@
+# project-services-list
+
+## Overview
+
+<!--mdtogo:Short-->
+
+The `project-services-list` function generates [project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service)
+resources from a list of services to enable [GCP APIs](https://cloud.google.com/apis) within a specified project.
+
+<!--mdtogo-->
+
+<!--mdtogo:Long-->
+
+## Usage
+
+`project-services-list` function can be used both declaratively and imperatively.
+
+```shell
+kpt fn eval --image gcr.io/kpt-fn/project-services-list:unstable --fn-config /tmp/services-config.yaml
+```
+
+The `project-services-list` function does the following:
+
+1. Generates [project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) resource
+for each service specified in the `spec.services` list.
+    * Adds all annotations defined for `ProjectServiceList` custom resource to each generated resource. This can be used for enabling features like
+[disable-on-destroy](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service#custom_resource_definition_properties) for generated services.
+    * Sets namespace if any defined for `ProjectServiceList` custom resource to each generated resource.
+    * Sets projectID if any defined for `ProjectServiceList` custom resource to each generated resource.
+1. Each generated [project service](https://cloud.google.com/config-connector/docs/reference/resource-docs/serviceusage/service) resource
+has a `blueprints.cloud.google.com/managed-by-project-services-list` annotation. This annotation allows `project-services-list` function to
+track generated resources for the declarative management of the generated resources. Any changes made to the generate resources will be overwritten and should be made to the `ProjectServiceList` CRD instead.
+
+### FunctionConfig
+
+This function only supports a custom resource `functionConfig` of kind `ProjectServiceList`.
+
+#### `ProjectServiceList`
+
+A functionConfig of kind `ProjectServiceList` has the following supported parameters:
+
+```yaml
+apiVersion: blueprints.cloud.google.com/v1alpha1
+kind: ProjectServiceList
+metadata:
+  name: my-project-services
+  annotations:
+    cnrm.cloud.google.com/deletion-policy: false
+spec:
+  services: # list of services to generate
+    - compute.googleapis.com
+  projectID: foo # explicit project ID to be set for the generated services
```

Comment:
Will it make sense to make it work better with `set-project-id` function ? for ex. adding the comment hints to make it work seamlessly with `set-project-id` ?

Reply:
Yup, sgtm. I think only change necessary is to add the `project-id` setter. `set-project-id` should then create pr update the value for `project-id` setter when its invoked.

Reply:
added project-id setter